vyper/parser/function_definitions/parse_public_function.py

```diff
@@ -0,0 +1,292 @@
+                        arg.arg
+                        for arg
+                        in sig.default_args
+                        if arg.arg not in current_sig_arg_names
+                    ]
+                    for arg_name in missing_arg_names:
+                        value = Expr(sig.default_values[arg_name], context).lll_node
+                        var = context.vars[arg_name]
+                        left = LLLnode.from_list(var.pos, typ=var.typ, location='memory',
+                                                 pos=getpos(code), mutable=var.mutable)
+                        set_defaults.append(make_setter(left, value, 'memory', pos=getpos(code)))
+
+                current_sig_arg_names = {x.name for x in default_sig.args}
+                base_arg_names = {arg.name for arg in sig.base_args}
+                copier_arg_count = len(default_sig.args) - len(sig.base_args)
+                copier_arg_names = current_sig_arg_names - base_arg_names
+
+                # Order copier_arg_names, this is very important.
+                copier_arg_names = {x.name for x in default_sig.args if x.name in copier_arg_names}
+
+                # Variables to be populated from calldata/stack.
+                default_copiers: List[Any] = []
+                if copier_arg_count > 0:
+                    # Get map of variables in calldata, with thier offsets
+                    offset = 4
+                    calldata_offset_map = {}
+                    for arg in default_sig.args:
+                        calldata_offset_map[arg.name] = offset
+                        offset += (
+                            32
+                            if isinstance(arg.typ, ByteArrayLike)
+                            else get_size_of_type(arg.typ) * 32
+                        )
+
+                    # Copy default parameters from calldata.
+                    dynamics: List[Any] = []
+                    for arg_name in copier_arg_names:
+                        var = context.vars[arg_name]
+                        calldata_offset = calldata_offset_map[arg_name]
+
+                        # Add clampers.
+                        default_copiers.append(make_arg_clamper(
+                            calldata_offset - 4,
+                            var.pos,
+                            var.typ,
+                        ))
+                        # Add copying code.
+                        _offset: Union[int, List[Any]] = calldata_offset
+                        if isinstance(var.typ, ByteArrayLike):
+                            _offset = ['add', 4, ['calldataload', calldata_offset]]
+                        default_copiers.append(get_public_arg_copier(
+                            memory_dest=var.pos,
+                            total_size=var.size * 32,
+                            offset=_offset,
+                        ))
+
+                    # Unpack byte array if necessary.
+                    if dynamics:
```

Comment:
dead block?

Reply:
Thanks! Yes this is irrelevant in the public context.